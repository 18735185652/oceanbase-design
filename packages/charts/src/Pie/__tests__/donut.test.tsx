import React from 'react';
import { render } from '@testing-library/react';
import DonutFloor from '../demo/donut-floor';

describe('donut', () => {
  it('floor number', () => {
    const { container, asFragment } = render(<DonutFloor />);
    expect(container.querySelectorAll('.g2-html-annotation')?.[1].textContent).toBe('21.16');
    expect(asFragment().firstChild).toMatchSnapshot();
  });
});
