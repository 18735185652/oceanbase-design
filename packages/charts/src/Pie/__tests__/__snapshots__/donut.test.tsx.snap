// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`donut floor number 1`] = `
<div
  data-chart-source-type="G2Plot"
  size-sensor-id="1"
  style="height: inherit;"
>
  <div
    style="position:relative;"
  >
    <canvas
      height="400"
      style="width: 400px; height: 400px; display: inline-block; vertical-align: middle;"
      width="400"
    />
    <div
      class="g2-html-annotation"
      style="position: absolute; pointer-events: none; width: 197.87573593128806px; transform: translate(-50%, -100%); overflow: hidden; white-space: nowrap; text-overflow: ellipsis; display: flex; justify-content: center; align-items: center; text-shadow:  0px 0px 0px; font-weight: 300; color: rgb(92, 107, 138); text-align: center; font-size: 14px; line-height: 1; font-family: Avenir-Heavy; left: 164.89644660940672px; top: 236px; z-index: 7;"
    >
      <div
        style="width:0px;;font-size:0em;line-height:1;"
      >
        总数
      </div>
    </div>
    <div
      class="g2-html-annotation"
      style="position: absolute; pointer-events: none; width: 197.87573593128806px; transform: translate(-50%, 0); overflow: hidden; white-space: nowrap; text-overflow: ellipsis; display: flex; justify-content: center; align-items: center; text-shadow:  0px 0px 0px; font-weight: bold; color: rgb(92, 107, 138); text-align: center; font-size: 32px; line-height: 1; font-family: PingFangSC; left: 164.89644660940672px; top: 182px; z-index: 7;"
    >
      <div
        style="width:0px;;font-size:0em;line-height:1;"
      >
        21.16
      </div>
    </div>
  </div>
  <object
    class="size-sensor-object"
    data="about:blank"
    style="display: block; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1; opacity: 0;"
    tabindex="-1"
    type="text/html"
  />
</div>
`;

exports[`donut statistic title and content formatter 1`] = `
<div
  data-chart-source-type="G2Plot"
  size-sensor-id="2"
  style="height: inherit;"
>
  <div
    style="position:relative;"
  >
    <canvas
      height="400"
      style="width: 400px; height: 400px; display: inline-block; vertical-align: middle;"
      width="400"
    />
    <div
      class="g2-html-annotation"
      style="position: absolute; pointer-events: none; width: 197.87573593128806px; transform: translate(-50%, -100%); overflow: hidden; white-space: nowrap; text-overflow: ellipsis; display: flex; justify-content: center; align-items: center; text-shadow:  0px 0px 0px; font-weight: 300; color: rgb(92, 107, 138); text-align: center; font-size: 14px; line-height: 1; font-family: Avenir-Heavy; left: 164.89644660940672px; top: 236px; z-index: 7;"
    >
      <div
        style="width:0px;;font-size:0em;line-height:1;"
      >
        custom title
      </div>
    </div>
    <div
      class="g2-html-annotation"
      style="position: absolute; pointer-events: none; width: 197.87573593128806px; transform: translate(-50%, 0); overflow: hidden; white-space: nowrap; text-overflow: ellipsis; display: flex; justify-content: center; align-items: center; text-shadow:  0px 0px 0px; font-weight: bold; color: rgb(92, 107, 138); text-align: center; font-size: 32px; line-height: 1; font-family: PingFangSC; left: 164.89644660940672px; top: 182px; z-index: 7;"
    >
      <div
        style="width:0px;;font-size:0em;line-height:1;"
      >
        custom content
      </div>
    </div>
  </div>
  <object
    class="size-sensor-object"
    data="about:blank"
    style="display: block; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1; opacity: 0;"
    tabindex="-1"
    type="text/html"
  />
</div>
`;
